python/pyarrow/array.pxi

```diff
@@ -3862,6 +3864,14 @@ cdef class StructArray(Array):
             Field py_field
             DataType struct_type
 
+        if fields is not None and structtype is not None:
+            raise ValueError('Must pass either fields or type, not both')
+        
```

Comment:
There seems to be empty spaces on this empty line that are making the linter fail

Reply:
nice catch, thank you!